### ISSUE TITLE
neon-runtime 0.4.1 breaks neon-serde

### ISSUE THREAD

**Issue by librelois** [neon-bindings/neon#612]
A cargo update broke my project because I use the [neon-serde](https://crates.io/crates/neon-serde) crate which depends on neon-runtime v0.4.

I get the following error:

```
error[E0061]: this function takes 3 arguments but 2 arguments were supplied
  --> /home/elois/.cargo/registry/src/github.com-1ecc6299db9ec823/neon-serde-0.4.0/src/errors.rs:96:13
   |
96 |             neon_runtime::error::throw_error_from_utf8(msg.as_ptr(), msg.len() as i32);
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ------------  ---------------- supplied 2 arguments
   |             |
   |             expected 3 arguments

error: aborting due to previous error

For more information about this error, try `rustc --explain E0061`.
error: could not compile `neon-serde`.
```

The throw_error_from_utf8 public function signature has been modified, it was necessary to upgrade to version 0.5 to comply with semver standards.
I had to manually modify my Cargo.lock to force the use of neon-runtime 0.4.0.

Now I can't do Cargo update anymore, and this is the case for any project that uses `neon-serde`.

Please release version 0.5.0 and yanked versions 0.4.1 and 0.4.2 to force cargo to use version 0.4.0.

**Comment by kjvalencik** [comment]
We will likely not be yanking this release. I am sympathetic that this may have cause some issues and we strive to keep neon semver compliant. However, neon-runtime is considered an _internal_ crate that should not be directly relied on; only the top level neon crate is intended to have a stable API.

We were unaware that the `neon-serde` crate relied on this internal function and are working with the author to fix it.

Again, I apologize for the inconvenience.

**Comment by kjvalencik** [comment]
Related issue https://github.com/GabrielCastro/neon-serde/issues/61

**Comment by kjvalencik** [comment]
@librelois `0.5.0` is released and `0.4.2` and `0.4.1` have been yanked.
